javalib/src/main/scala/java/io/ByteArrayInputStream.scala

```diff
@@ -39,6 +39,18 @@ class ByteArrayInputStream(
     }
   }
 
+  def readNBytes(b: Array[Byte], off: Int, len: Int): Int = {
+    val n = read(b, off, len)
+    if (n == -1) 0 else n
+  }
+
+  def readNBytes(len: Int): Array[Byte] = {
+    val n = if (buf.length > len) len else buf.length
```

Comment:
By my reading of the Java 11 spec for this method, this
line is out-of-spec. I recommend deletion and let the
local read(3args) method handle the Exception.
 ``` IndexOutOfBoundsException - If off is negative, len is negative, or len is greater than b.length - off```

That would also allow this implementation becoming an `InputStream` default method.